### PR TITLE
JDK-8310118: Resource files should be moved to appropriate directories

### PR DESCRIPTION
**Pull request by hns**
Please review a change to rename the directories previoiusly called `resources` and `script-dir` in `javadoc`-generated documentation to `resource-files` and `script-files`, and consistently moving all files that are not related to the documented code into these directories. This includes scripts and stylesheets added via command line options.

In the process I cleaned up the code which copies scripts and resources. I added a new private `HtmlDoclet.copyResource` method which takes care of resolving resources relative to the `doclet.formats.html` or `doclet.toolkit` packages and determines whether a resource needs localization based on the file extension of the source resource. The new code will also throw an exception if the source resource does not exist, while previously `DocFile.copyResource` returned silently. 

Another unrelated cleanup was to remove a bunch of jQuery-UI image files in a resource subdirectory that were not used anymore.

In addition to make sure all tests pass I did extensive manual testing of the generated docs to make sure all scripts and resources are loaded correctly. This includes the `@import` statement for `resources/fonts/dejavu.css` in the default stylesheet, which is now changed to `fonts/dejavu.css` assuming the `fonts` directory will now be located in `resource-files`.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8310118](https://bugs.openjdk.org/browse/JDK-8310118): Resource files should be moved to appropriate directories (**Bug** - P4)

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to [7995454c](https://git.openjdk.org/jdk/pull/15180/files/7995454c5f5d30dd7bb47062a817dff21bb66777)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15180/head:pull/15180` \
`$ git checkout pull/15180`

Update a local copy of the PR: \
`$ git checkout pull/15180` \
`$ git pull https://git.openjdk.org/jdk.git pull/15180/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15180`

View PR using the GUI difftool: \
`$ git pr show -t 15180`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15180.diff">https://git.openjdk.org/jdk/pull/15180.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15180#issuecomment-1668131649)

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDoclet.java [hunk 1]
```diff
@@ -444,7 +421,29 @@ public Set<? extends Option> getSupportedOptions() {
...
 
-    private void performCopy(String filename, DocPath targetPath) throws DocFileIOException {
+    private void copyResource(DocPath sourcePath, DocPath targetPath, boolean replaceNewLine)
+            throws DocletException {
+        DocPath resourcePath = DocPaths.RESOURCES.resolve(sourcePath);
+        // Look for resource in formats/html resources first
+        URL resourceURL = HtmlConfiguration.class.getResource(resourcePath.getPath());
+        // If not found try toolkit resources
+        if (resourceURL == null) {
+            resourceURL = BaseConfiguration.class.getResource(resourcePath.getPath());
```
Comment (jonathan-gibbons) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDoclet.java:431]: It always seemed wrong (a bug) that `stylesheet.css` was down in the `toolkit` world.
It is also possible to reason that `script.js` should not be in the `toolkit` world either.
Maybe this would be a good time to move those files into the `formats/html` world, as
well as the two `.svg` files, so that all resource files that need to be copied into the 
generated API directories are in the `formats/html` world.

In other words, we might be able to avoid lines 430-432

Comment (hns) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDoclet.java:431]: Yes, that occurred to me as well. I'll move the files and update the code.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDoclet.java [hunk 2]
```diff
@@ -304,31 +301,25 @@ protected void generateOtherFiles(ClassTree classTree)
...
-        f.copyResource(DocPaths.RESOURCES.resolve(DocPaths.CLIPBOARD_SVG), true, true);
-        f = DocFile.createFileForOutput(configuration, DocPaths.LINK_SVG);
-        f.copyResource(DocPaths.RESOURCES.resolve(DocPaths.LINK_SVG), true, true);
-        if (options.createIndex()) {
-            f = DocFile.createFileForOutput(configuration, DocPaths.SEARCH_JS);
-            f.copyResource(DOCLET_RESOURCES.resolve(DocPaths.SEARCH_JS_TEMPLATE), configuration.docResources);
-
-            f = DocFile.createFileForOutput(configuration, DocPaths.SEARCH_PAGE_JS);
-            f.copyResource(DOCLET_RESOURCES.resolve(DocPaths.SEARCH_PAGE_JS), configuration.docResources);
+        copyResource(DocPaths.SCRIPT_JS, DocPaths.SCRIPT_FILES.resolve(DocPaths.SCRIPT_JS), true);
```
Comment (jonathan-gibbons) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDoclet.java:307]: In general, in this method, there is a lot of code of the form `DocPaths.SOME_DIR.resolve(DocPaths.SOME_FILE)`.

In general, I think it is a side-effect of prior "bad" design that these files used to be in the root directory of the generated API, resulting in it being sufficient to define just a simple filename in `DocPaths`.

In general, I think it would be better to move these `resolve` calls into `DocPaths`, so that it becomes a single reference (i.e. without any `.resolve`) here in this code.

As an example, I grepped for SCRIPT_JS:
```
open/src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDoclet.java:        copyResource(DocPaths.SCRIPT_JS, DocPaths.SCRIPT_FILES.resolve(DocPaths.SCRIPT_JS), true);
open/src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/Head.java:            addScriptElement(head, DocPaths.SCRIPT_FILES.resolve(DocPaths.SCRIPT_JS));
open/src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/DocPaths.java:    public static final DocPath SCRIPT_JS = DocPath.create("script.js");
```

In both use sites, the usage is: `DocPaths.SCRIPT_FILES.resolve(DocPaths.SCRIPT_JS)` suggesting that the declaration should really be something like:
```
open/src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/DocPaths.java:    public static final DocPath SCRIPT_JS = SCRIPT_FILES.resolve("script.js");
```

**Review by jonathan-gibbons — APPROVED** [review]
I wish there was a way to reduce the number of `.resolve` calls in `HtmlDoclet.init`, since they do clutter up the code a bit.

However, I don't have any simple fixups, and such suggestions that I have would be a bigger change than is probably warranted here. For example,
```
   copyResources(Path toDir, String... resources)
```

Maybe next time, unless you're feeling keen and eager now.

Approved by default.

**Comment by hns** [comment]
> 
> ```
>    copyResources(Path toDir, String... resources)
> ```
> 

I would have liked to simplify the code further, but apart from different destinations, there's also different handling of newline character replacement (even within the same file type), so the resulting code would not be much more compact than the current 20 lines.
